packages/ag-charts-enterprise/src/gradient-legend/gradientLegend.ts

```diff
@@ -1,77 +1,171 @@
-    NUMBER,
-    OPT_BOOLEAN,
-    OPT_NUMBER,
-    OPT_FUNCTION,
-    OPT_FONT_STYLE,
-    OPT_FONT_WEIGHT,
-    POSITION,
-    STRING,
-} = _ModuleSupport;
-const { BBox, Group, Rect, LinearGradientFill, Selection, Text, Triangle } = _Scene;
-const { createId } = _Util;
-
-class GradientLegendLabel {
-    @Validate(OPT_NUMBER(0))
-    maxLength?: number = undefined;
-
-    @Validate(COLOR_STRING)
-    color: string = 'black';
-
-    @Validate(OPT_FONT_STYLE)
-    fontStyle?: FontStyle = undefined;
-
-    @Validate(OPT_FONT_WEIGHT)
-    fontWeight?: FontWeight = undefined;
 
+const { AND, BOOLEAN, Layers, NUMBER, OPT_BOOLEAN, POSITION, Validate, Default, GREATER_THAN, NUMBER_OR_NAN } =
+    _ModuleSupport;
+const { BBox, Group, Rect, LinearGradientFill, Triangle } = _Scene;
+const { createId, Logger, DeprecatedAndRenamedTo } = _Util;
+
+class GradientBar {
     @Validate(NUMBER(0))
-    fontSize: number = 12;
+    thickness = 16;
 
-    @Validate(STRING)
-    fontFamily: string = 'Verdana, sans-serif';
+    @Validate(NUMBER(0))
+    preferredLength = 100;
+}
 
-    @Validate(OPT_FUNCTION)
-    formatter?: (params: AgChartLegendLabelFormatterParams) => string = undefined;
+class GradientLegendAxisTick extends _ModuleSupport.AxisTick<_Scale.LinearScale, number> {
+    override enabled = false;
+    // TODO: Set tickSize to zero when tick.enabled = false in axis.ts
+    // https://ag-grid.atlassian.net/browse/AG-10059
+    override size = 0;
+
+    @Validate(AND(NUMBER_OR_NAN(1), GREATER_THAN('minSpacing')))
+    @Default(NaN)
+    override maxSpacing: number = NaN;
 }
 
-class GradientLegendStop {
-    readonly label = new GradientLegendLabel();
-    @Validate(NUMBER(0))
-    padding = 8;
+class GradientLegendAxis extends _ModuleSupport.CartesianAxis<_Scale.LinearScale, number> {
+    colorDomain: number[] = [];
+
+    get interval(): _ModuleSupport.AxisTick<_Scale.LinearScale> {
+        return this.tick;
+    }
+
+    constructor(ctx: _ModuleSupport.ModuleContext) {
+        super(ctx, new _Scale.LinearScale());
+        this.nice = false;
+        this.line.enabled = false;
+    }
+
+    override calculateDomain() {
+        this.dataDomain = this.normaliseDataDomain(this.colorDomain);
+    }
+
+    override formatDatum(datum: number): string {
+        if (typeof datum === 'number') {
+            return datum.toFixed(2);
+        } else {
+            Logger.warnOnce(
+                'data contains Date objects which are being plotted against a number axis, please only use a number axis for numbers.'
+            );
+            return String(datum);
+        }
+    }
+
+    protected override createTick() {
+        return new GradientLegendAxisTick();
+    }
 }
 
-class GradientBar {
-    @Validate(NUMBER(0))
-    thickness = 16;
+class GradientLegendInterval implements AgGradientLegendIntervalOptions {
+    constructor(private tick: GradientLegendAxisTick) {}
 
-    @Validate(NUMBER(0))
-    preferredLength = 100;
+    get enabled() {
```

Comment:
Maybe use the `@ProxyPropertyOnWrite` decorator to clean up the use of `set`/`get` pairs here?
https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-community/src/chart/caption.ts#L37